tests/e2e/PreMutantFilter/filter/ExampleFilter.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace PreMutantFilter;
+
+use Infection\Mutant\Mutant;
+use Pipeline\Interfaces\StandardPipeline;
+
+class ExampleFilter
+{
+    private $count;
+
+    public function __construct(StandardPipeline $mutants)
```

Comment:
The ctor used as interception point should probably be avoided

Reply:
What would you prefer? 

Current approach makes possible to use a filter with `infection.phar` because it requires no imported classes. You don't even have to depend/import Mutant/Pipeline. If we'd add an interface like `InfectionPlugin`, could we still use it with `.phar`?

I might not know if Box can exclude certain interfaces from being sub-namespaces. @theofidry Is there such a thing?

Reply:
See the review namespace at https://github.com/infection/infection/pull/1327#pullrequestreview-484236491

As for the namespace, I'd suggest whitelisting any non-`@internal` symbols involved later on. I personally never use PHARs, since they create very artifical problems, like the one you just mentioned (symbol exists, but is not really visible outside the scope of a phar binary execution)